tests/sentry/workflow_engine/migration_helpers/test_migrate_alert_rule.py

```diff
@@ -0,0 +1,83 @@
+from sentry.incidents.grouptype import MetricAlertFire
+from sentry.snuba.models import QuerySubscription
+from sentry.testutils.cases import APITestCase
+from sentry.users.services.user.service import user_service
+from sentry.workflow_engine.migration_helpers.alert_rule import migrate_alert_rule
+from sentry.workflow_engine.models import (
+    AlertRuleDetector,
+    AlertRuleWorkflow,
+    DataSource,
+    DataSourceDetector,
+    Detector,
+    DetectorState,
+    DetectorWorkflow,
+    Workflow,
+    WorkflowDataConditionGroup,
+)
+from sentry.workflow_engine.types import DetectorPriorityLevel
+
+
+class AlertRuleMigrationHelpersTest(APITestCase):
+    def setUp(self):
+        self.metric_alert = self.create_alert_rule()
+        self.rpc_user = user_service.get_user(user_id=self.user.id)
+
+    def test_create_metric_alert(self):
+        """
+        Test that when we call the helper methods we create all the ACI models correctly for an alert rule
+        """
+        migrate_alert_rule(self.metric_alert, self.rpc_user)
+
+        alert_rule_workflow = AlertRuleWorkflow.objects.get(alert_rule=self.metric_alert)
+        alert_rule_detector = AlertRuleDetector.objects.get(alert_rule=self.metric_alert)
+
+        workflow = Workflow.objects.get(id=alert_rule_workflow.workflow.id)
+        assert workflow.name == self.metric_alert.name
+        assert self.metric_alert.organization
+        assert workflow.organization_id == self.metric_alert.organization.id
+        detector = Detector.objects.get(id=alert_rule_detector.detector.id)
+        assert detector.name == self.metric_alert.name
+        assert detector.project_id == self.project.id
+        assert detector.enabled is True
+        assert detector.description == self.metric_alert.description
+        assert detector.owner_user_id == self.metric_alert.user_id
+        assert detector.owner_team == self.metric_alert.team
+        assert detector.type == MetricAlertFire.slug
+        assert detector.config == {
+            "threshold_period": self.metric_alert.threshold_period,
+            "sensitivity": None,
+            "seasonality": None,
+            "comparison_delta": None,
+        }
+
+        detector_workflow = DetectorWorkflow.objects.get(detector=detector)
+        assert detector_workflow.workflow == workflow
+
+        workflow_data_condition_group = WorkflowDataConditionGroup.objects.get(workflow=workflow)
```

Comment:
i think the when condition group doesn't need to exist in `WorkflowDataConditionGroup`, it's already attached to the workflow via workflow.when_condition_group. i believe this model is used for the IF condition groups